### ISSUE TITLE
SecurityException crash in hyper-sdk-react 3.0.6

### ISSUE THREAD

**Issue by praveen-suggaa** [juspay/hyper-sdk-react#58]
We are experiencing a crash in our app in production that appears to be caused by the hyper-sdk-react library. The crash occurs when attempting to register a receiver internally within the SmsConsentHandler class.

```
Exception java.lang.SecurityException:
  at android.os.Parcel.createExceptionOrNull (Parcel.java:3069)
  at android.os.Parcel.createException (Parcel.java:3053)
  at android.os.Parcel.readException (Parcel.java:3036)
  at android.os.Parcel.readException (Parcel.java:2978)
  at android.app.IActivityManager$Stub$Proxy.registerReceiverWithFeature (IActivityManager.java:6157)
  at android.app.ContextImpl.registerReceiverInternal (ContextImpl.java:1913)
  at android.app.ContextImpl.registerReceiver (ContextImpl.java:1853)
  at android.app.ContextImpl.registerReceiver (ContextImpl.java:1841)
  at android.content.ContextWrapper.registerReceiver (ContextWrapper.java:772)
  at in.juspay.hypersdk.core.SmsConsentHandler.lambda$new$0$in-juspay-hypersdk-core-SmsConsentHandler
  at in.juspay.hypersdk.core.SmsConsentHandler$$ExternalSyntheticLambda3.run
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1145)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:644)
  at java.lang.Thread.run (Thread.java:1012)
Caused by android.os.RemoteException: Remote stack trace:
  at com.android.server.am.ActivityManagerService.registerReceiverWithFeature (ActivityManagerService.java:16680)
  at android.app.IActivityManager$Stub.onTransact$registerReceiverWithFeature$ (IActivityManager.java:11613)
  at android.app.IActivityManager$Stub.onTransact (IActivityManager.java:2961)
  at com.android.server.am.ActivityManagerService.onTransact (ActivityManagerService.java:3199)
  at android.os.Binder.execTransactInternal (Binder.java:1375)
```

We are using hyper-sdk-react version `3.0.6`

This issue is similar to one previously reported and fixed in hyper-sdk-flutter `4.0.6` but there was no release for the same fix for hyper-sdk-react. Please see this [GitHub issue](https://github.com/juspay/hyper-sdk-flutter/issues/14) for reference.

**Comment by praveen-suggaa** [comment]
Any update on this?

**Comment by yaswanth6240** [comment]
Hey @praveen-suggaa this crash was fixed in hyper-sdk-react version **3.0.17**.
